crates/sui-indexer-alt/src/handlers/coin_balance_buckets_pruner.rs

```diff
@@ -6,22 +6,62 @@ use std::{collections::BTreeMap, sync::Arc};
 use anyhow::Result;
 use diesel::sql_query;
 use diesel_async::RunQueryDsl;
+use sui_field_count::FieldCount;
 use sui_indexer_alt_framework::pipeline::{concurrent::Handler, Processor};
 use sui_pg_db as db;
-use sui_types::full_checkpoint_content::CheckpointData;
+use sui_types::{base_types::ObjectID, full_checkpoint_content::CheckpointData};
 
-use super::coin_balance_buckets::{
-    CoinBalanceBucketChangeKind, CoinBalanceBuckets, ProcessedCoinBalanceBucket,
-};
+use super::coin_balance_buckets::{get_coin_balance_bucket, get_coin_owner};
 
 pub(crate) struct CoinBalanceBucketsPruner;
 
+pub(crate) struct CoinBalanceBucketToBePruned {
+    pub object_id: ObjectID,
+    pub cp_sequence_number_exclusive: u64,
+}
+
 impl Processor for CoinBalanceBucketsPruner {
     const NAME: &'static str = "coin_balance_buckets_pruner";
-    type Value = ProcessedCoinBalanceBucket;
+    type Value = CoinBalanceBucketToBePruned;
 
     fn process(&self, checkpoint: &Arc<CheckpointData>) -> Result<Vec<Self::Value>> {
-        CoinBalanceBuckets.process(checkpoint)
+        let cp_sequence_number = checkpoint.checkpoint_summary.sequence_number;
+        let checkpoint_input_objects = checkpoint.checkpoint_input_objects();
+        let latest_live_output_objects: BTreeMap<_, _> = checkpoint
+            .latest_live_output_objects()
+            .into_iter()
+            .map(|o| (o.id(), o))
+            .collect();
+        let mut values = Vec::with_capacity(checkpoint_input_objects.len());
```

Comment:
Is the `with_capacity` necessary here? Do we expect most of the input objects to be coins?